test/functions.jl

```diff
@@ -44,6 +44,18 @@ using Statistics
         end
     end
 
+    @testset "$f!" for (f,f!) in zip((sum, prod, maximum, minimum), (sum!, prod!, maximum!, minimum!))
+        nda1 = sum(nda, dims=1)
+        nda2 = sum(nda, dims=2)
```

Comment:
Can it be made more clear in this test when we are testing a column vector vs row vector etc.
I think at least it is woth declaring all things inside this testset, 
rather than using ones declared earlier.

(I've been tranistioning more testsets over to being self contained like that )